codice/ddf
# [2.11.x] DDF-3280 Upgrade Karaf to 4.1.4

#### What does this PR do?
Initial PR to update Karaf from 4.1.2 to 4.1.4

#### Who is reviewing it?
@emmberk @peterhuffer @oconnormi  @vinamartin

#### Choose 2 committers to review/merge the PR.
(please choose ONLY two committers from below, delete the rest)

@coyotesqrl
@lessarderic
@rzwiefel
@stustison

#### How should this be tested? (List steps with links to updated documentation)
 Manually test basic functionality of DDF

The following have been tested but could be tested again if desired:

- Drop .cfg and .config files in the /etc directory, and confirm that the configurations are installed successfully.
-  Confirm that the bin scripts work as expected.
-  Walkthrough the Hardening Guide. 
-  Confirm that the waitForReady command still works.
-  Confirm that installing via each install profile works.
-  Confirm that logging still works and is still configurable from org.ops4j.pax.logging.cfg and from the Karaf console. Also, try switching the logging configuration to the xml-style configuration.
-  Test that the common console commands are still functional.
-  Confirm SSHing into a running instance still works. Confirm SSHing into itests still works.
-  Confirm debugging a live system
-  Confirm that the Karaf instance can be restarted.

#### Any background context you want to provide?
Karaf change set https://github.com/apache/karaf/compare/karaf-4.1.2...karaf-4.1.4

#### What are the relevant tickets?
https://codice.atlassian.net/browse/DDF-3280
